### PR TITLE
feat: add local-dns command

### PR DESCRIPTION
**Pull request by augustoccesar**
### Description
Makes it possible to install local DNS routing for linkup.

Uses of a combination of [caddy](https://caddyserver.com/) and [dnsmasq](https://thekelleys.org.uk/dnsmasq/doc.html) to achieve local TLS/SSL and DNS for this to work in a frictionless way.

### CODE REVIEW

File: linkup-cli/src/background_tunnel.rs [hunk 1]
```diff
@@ -6,10 +6,11 @@ use std::thread;
 use std::time::Duration;
 
 use daemonize::{Daemonize, Outcome};
+use nix::sys::signal::Signal;
```
Comment (ostenbom) [inline linkup-cli/src/background_tunnel.rs:9]: Wrong signal? Should it be nix?

Comment (augustoccesar) [reply linkup-cli/src/background_tunnel.rs:9]: Well, this is the currently one in master, so I thought so 😅 
https://github.com/mentimeter/linkup/blob/main/linkup-cli/src/signal.rs

But I'm not quite sure 🤔

File: linkup-cli/src/local_dns.rs [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    process::{Command, Stdio},
+};
+
+use crate::{
+    linkup_file_path,
+    local_config::{config_path, get_config},
+    services, CliError, Result, LINKUP_CF_TLS_API_ENV_VAR, LINKUP_LOCALDNS_INSTALL,
+};
+
+pub fn install(config_arg: &Option<String>) -> Result<()> {
```
Comment (ostenbom) [inline linkup-cli/src/local_dns.rs:12]: One thing we might consider doing here is making sure we're on macos? I'm not sure what the linux support is like 😅

File: linkup-cli/src/stop.rs [hunk 3]
```diff
@@ -27,13 +29,17 @@ pub fn stop() -> Result<(), CliError> {
...
-    let tunnel_stopped = stop_pid_file(LINKUP_CLOUDFLARED_PID);
-    if tunnel_stopped.is_ok() {
-        let _ = std::fs::remove_file(linkup_file_path(LINKUP_CLOUDFLARED_PID));
+    let local_stopped = stop_pid_file(
+        &linkup_file_path(LINKUP_LOCALSERVER_PID_FILE),
+        Signal::SIGINT,
+    );
+
+    // TODO(augustoccesar)[2023-09-26]: This pidfile seems to have the liknkup process pid, not cloudflared pid. So this is deleting the file
+    //   but not actually stopping cloudflared
```
Comment (ostenbom) [inline linkup-cli/src/stop.rs:38]: Don't need this comment any more, works right?

**Review by ostenbom — APPROVED** [review]

File: linkup-cli/src/local_dns.rs (hunk 2 above)
Comment (augustoccesar) [reply linkup-cli/src/local_dns.rs:12]: Hmmm, that is a good point. We can take a look it as well!
I guess we can find someone with a Linux and try it out as well 👀

Comment (ostenbom) [reply linkup-cli/src/local_dns.rs:12]: It's more of a nice to have, we can take a look at it later!

**Pull request merged**